src/firebase/index.js

```diff
@@ -0,0 +1,35 @@
+import firebase from "firebase/compat/app";
+import * as firebaseui from "firebaseui";
+import "firebase/auth";
+import "firebase/database";
+
+// Your web app's Firebase configuration
+const config = {
+  apiKey: "AIzaSyB3_ktYCDAvZR8ViEgmkKT8CDMPm1Q53Zo",
+  authDomain: "safetydungeon.firebaseapp.com",
+  projectId: "safetydungeon",
+  storageBucket: "safetydungeon.appspot.com",
+  messagingSenderId: "156587932337",
+  appId: "1:156587932337:web:116ae943f4f2ff92790090",
+};
+
+const app = firebase.initializeApp(config);
+const auth = firebase.auth();
+//const db = app.database();
+
+const authUiConfig = {
+  signInSuccessUrl: "/",
+  signInOptions: [
+    {
+      provider: firebase.auth.EmailAuthProvider.PROVIDER_ID,
+      customParameters: {
+        prompt: "select_account",
+      },
+    },
+  ],
+};
+
+const authUi = new firebaseui.auth.AuthUI(auth);
+
+export default app;
```

Comment:
One thing I really don't understand: why is "export default app" here. App isn't originating in this component...

Does this mean - export this to "app" by default? So app doesn't have to explicitly import it?

Or does it define a hierarchy of components and this pointing upwards?

Reply:
> App isn't originating in this component...

Confusing naming maybe. `app` _is_ originating here on L16. 

> Does this mean - export this to "app" by default? So app doesn't have to explicitly import it?
> Or does it define a hierarchy of components and this pointing upwards?

I think neither (if I understand your comments).

We can have a play with this with some code maybe. For now the docs are here: https://developer.mozilla.org/en-US/docs/web/javascript/reference/statements/export which we can refer back to later.